gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/SubscriptionServiceImpl.java

```diff
@@ -987,31 +991,42 @@ public SubscriptionEntity resumeConsumer(final ExecutionContext executionContext
             validateConsumerStatus(subscription, genericApiModel);
 
             if (subscription.canBeStartedByConsumer()) {
-                Subscription previousSubscription = new Subscription(subscription);
-                final Date now = new Date();
-                subscription.setUpdatedAt(now);
-                subscription.setConsumerPausedAt(null);
-                subscription.setConsumerStatus(Subscription.ConsumerStatus.STARTED);
+                return updateSubscription(executionContext, subscription, apiId);
+            }
 
-                subscription = subscriptionRepository.update(subscription);
+            throw new SubscriptionNotPausedException(subscription);
+        } catch (TechnicalException ex) {
+            throw new TechnicalManagementException(
+                String.format("An error occurs while trying to resume subscription %s", subscriptionId),
+                ex
+            );
+        }
+    }
 
-                createAudit(
-                    executionContext,
-                    apiId,
-                    subscription.getApplication(),
-                    SUBSCRIPTION_RESUMED_BY_CONSUMER,
-                    subscription.getUpdatedAt(),
-                    previousSubscription,
-                    subscription
-                );
+    @Override
+    public SubscriptionEntity resumeFailure(final ExecutionContext executionContext, String subscriptionId) {
```

Comment:
I am not sure to understand why you need this new resumeFailure method (and Resource) 
imho it is already the purpose of `resumeConsumer()` method, even if it might need some adaptation

Reply:
Agreed but after discussion with Ouahid we decided to move it to the separate endpoint